koku/masu/database/trino_sql/gcp/openshift/reporting_ocpgcpcostlineitem_daily_summary.sql

```diff
@@ -303,8 +303,22 @@ SELECT gcp.uuid as gcp_uuid,
     max(nullif(gcp.sku_id, '')) as sku_id,
     max(nullif(gcp.sku_description, '')) as sku_alias,
     max(nullif(gcp.location_region, '')) as region,
-    max(gcp.usage_pricing_unit) as unit,
-    cast(max(gcp.usage_amount_in_pricing_units) AS decimal(24,9)) as usage_amount,
+    CASE
+        WHEN max(usage_pricing_unit) = 'hour'
+        THEN 'Hrs'
+        WHEN max(usage_pricing_unit) = 'gibibyte'
+        THEN 'GB'
+        WHEN max(usage_pricing_unit) = 'gibibyte month'
+        THEN 'GB-Mo'
+        WHEN max(usage_pricing_unit) = 'gibibyte hour'
+        THEN 'GB-Hours'
+    ELSE max(usage_pricing_unit)
+    END as unit,
+    CASE
+        WHEN max(usage_pricing_unit) = 'gibibyte month' OR max(usage_pricing_unit) = 'gibibyte' OR max(usage_pricing_unit) = 'gibibyte hour'
+        THEN cast(sum(usage_amount_in_pricing_units) * 1.073741824 AS decimal(24,9)) -- Convert to gigabyte
```

Comment:
So, one thing we want to pay special attention to here is how we are calculating the previous values. Notice how original SQL is using a `max` here:
```
cast(max(gcp.usage_amount_in_pricing_units) AS decimal(24,9)) as usage_amount,
```
But you changed it to a sum to match what you did in the gcp cost entry table:
```
cast(sum(gcp.usage_amount_in_pricing_units) AS decimal(24,9)) as usage_amount,
```
That could result in an unintended bloating of the number:
```
 usage_unit |  original_amount   |    usage_amount_in     |      usage_amount
------------+--------------------+------------------------+------------------------
 GB-Mo      |      244.725077280 |       173704.559606529 |       173704.559606529
 Hrs        |        0.997722222 |          266.195118511 |          266.195118511
 GB         | 92993765.703750000 | 140059886352.765350000 | 140059886352.765350000
 GB-Hours   |        0.335236414 |          130.238963714 |          130.238963714
 ```
 My guess is that the `gcp_openshift_daily` is already an aggregated value, and we might be doing some double aggregation.

Reply:
Makes sense, but do you suggest switching back to **max**? Because in the main branch, it is currently using **max**.

Reply:
Yes! I recommend switching back to max for the ocp on gcp flow.

Reply:
@bacciotti I figured out a better way to explain this! I set up a meeting for next week to talk through our daily tables & Trino managed tables. I want to talk through how we create the parquet files for the daily cloud tables.  I think it will help explain why we use max here instead of sum.